main.cpp

```diff
@@ -1 +1,307 @@
+#include <iostream> //this is only for cout and tolower
+#include <stdlib.h>
+#include <string.h>
 
+
+// Initializes list
+// void StringListInit(char*** list); 
+void StringListInit(char*** list)
+{
+    if (*list) return;
+    *list = (char**)calloc(2, sizeof(char*));   
+}
+
+// Destroy list and set pointer to NULL.
+// void StringListDestroy(char*** list);
+void StringListDestroy(char*** list)
+{
+    if (!*list) return;
+    char** current(*list), **previous(NULL);
+    
+    while(current[1])
+    {
+        previous = current;
+        current = (char**)current[1];
+        free(previous[0]);
+        free(previous);
+    }
+    free(current[0]);
+    free(current);
+    
+    *list = NULL;
+    list[0] = NULL;
+}
+
+// Inserts value at the end of the list.
+// void StringListAdd(char*** list, String str);
+void StringListAdd(char*** head, const char* val)
+{
+    if (!val[0] && !val) return;
+
+    if (!(*head)[0] && !(*head)[1])
+    {
+        (*head)[0] = (char*)calloc(strlen(val) + 1, sizeof(char));
+        if ((*head)[0]) strcpy((*head)[0], val);
+        (*head)[1] = NULL;
+    }
+
+    else 
+    {
+        char** temp;
+
+        temp = *head;
+        
+        while (temp[1]!=NULL) 
+        {
+            temp = (char **)temp[1];
+        }
+        
+        char** node = (char**)calloc(2, sizeof(char *));
+
+        node[0] = (char*)calloc(strlen(val) + 1, sizeof(char));
+        if (node[0]) strcpy(node[0], val);
+        node[1] = NULL;
+
+        temp[1] = (char *)node;
+    }
+}
+
+void display(char** head)
+{
+    char **temp;
+
+    temp = head;
+    if(!temp) printf("List is empty\n");
+    else
+    {
+        printf("display:\n");
+        while (temp!=NULL) 
+        {
+            printf("%s\n", temp[0]);
+            temp = (char **)temp[1];
+        }
+        printf("\n");
+    }
+}
+
+//Additional
+//Removes node from tail
+void popBack(char*** head)
+{
+    if(!*head) return;
+
+    char** current(*head), **previous(NULL);
+    
+    if(!current[1])
+    {
+        free(current[0]);
+        free(current);
+        *head = NULL;
+        head[0] = NULL;
+    }
+
+    else
+    {
+        while(current[1]!=NULL)
+        {
+            previous = current;
+            current = (char**)current[1];
+        }
+        previous[1] = NULL;
+        free(current[0]);
+        free(current);
+       
+    }
+}
+
+// Returns the number of items in the list.
+// int StringListSize(char** list);  
+int StringListSize (char** list)
+{
+    if(!list) return 0;
+
+    int counter(0);
```

Comment:
C language does not allow to initialize 'int' type as C++ copy constructor. It is error